Compiling ERROR

I try to run a test example run_node 0 with both normal rust and 1.31.0-nightly
but when compiling, a number of errors appear:
1. firstly, in Cargo.toml there is no existing tokio branch,
2. if you uncomment the previous version of tokio:

```
   Compiling hydrabadger v0.1.0 (/home/user/hydrabadger-master)                                                                         
error[E0599]: no method named `is_empty` found for type `std::result::Result<hbbft::sync_key_gen::AckOutcome, hbbft::sync_key_gen::Error>` in the current scope
   --> src/hydrabadger/handler.rs:197:23                                                                                                
    |                                                                                                                                   
197 |         if !fault_log.is_empty() {                                                                                                
    |                       ^^^^^^^^                                                                                                    
                                                                                                                                        
error[E0277]: `hbbft::sync_key_gen::AckOutcome` doesn't implement `std::fmt::Debug`                                                     
   --> src/hydrabadger/handler.rs:198:63                                                                                                
    |                                                                                                                                   
198 |             error!("Errors handling ack: '{:?}':\n{:?}", ack, fault_log);                                                         
    |                                                               ^^^^^^^^^ `hbbft::sync_key_gen::AckOutcome` cannot be formatted using `{:?}` because it doesn't implement `std::fmt::Debug`
    |                                                                                                                                   
    = help: the trait `std::fmt::Debug` is not implemented for `hbbft::sync_key_gen::AckOutcome`                                        
    = note: required because of the requirements on the impl of `std::fmt::Debug` for `std::result::Result<hbbft::sync_key_gen::AckOutcome, hbbft::sync_key_gen::Error>`
    = note: required by `std::fmt::Debug::fmt`                                                                                          
                                                                                                                                        
error[E0308]: mismatched types                                                                                                          
   --> src/hydrabadger/handler.rs:237:21                                                                                                
    |                                                                                                                                   
237 |                     Some(PartOutcome::Valid(ack)) => ack,                                                                         
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `std::result::Result`, found enum `std::option::Option`           
    |                                                                                                                                   
    = note: expected type `std::result::Result<hbbft::sync_key_gen::PartOutcome, hbbft::sync_key_gen::Error>`                           
               found type `std::option::Option<_>`                                                                                      
                                                                                                                                        
error[E0308]: mismatched types                                                                                                          
   --> src/hydrabadger/handler.rs:238:21                                                                                                
    |                                                                                                                                   
238 |                     Some(PartOutcome::Invalid(faults)) => panic!(                                                                 
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `std::result::Result`, found enum `std::option::Option`      
    |                                                                                                                                   
    = note: expected type `std::result::Result<hbbft::sync_key_gen::PartOutcome, hbbft::sync_key_gen::Error>`                           
               found type `std::option::Option<_>`                                                                                      
                                                                                                                                        
error[E0308]: mismatched types                                                                                                          
   --> src/hydrabadger/handler.rs:243:21                                                                                                
    |                                                                                                                                   
243 |                     None => {                                                                                                     
    |                     ^^^^ expected enum `std::result::Result`, found enum `std::option::Option`                                    
    |                                                                                                                                   
    = note: expected type `std::result::Result<hbbft::sync_key_gen::PartOutcome, hbbft::sync_key_gen::Error>`                           
               found type `std::option::Option<_>`                                                                                      
                                                                                                                                        
error[E0308]: mismatched types                                                                                                          
   --> src/hydrabadger/handler.rs:252:61                                                                                                
    |                                                                                                                                   
252 |                 ack_queue.as_ref().unwrap().push((*src_uid, ack.clone()));                                                        
    |                                                             ^^^^^^^^^^^ expected struct `hbbft::sync_key_gen::Ack`, found enum `std::option::Option`
    |                                                                                                                                   
    = note: expected type `hbbft::sync_key_gen::Ack`                                                                                    
               found type `std::option::Option<hbbft::sync_key_gen::Ack>`                                                               
                                                                                                                                        
error[E0308]: mismatched types                                                                                                          
   --> src/hydrabadger/handler.rs:261:71                                                                                                
    |                                                                                                                                   
261 |                 self.wire_to_validators(WireMessage::key_gen_part_ack(ack), &peers);                                              
    |                                                                       ^^^ expected struct `hbbft::sync_key_gen::Ack`, found enum `std::option::Option`
    |                                                                                                                                   
    = note: expected type `hbbft::sync_key_gen::Ack`                                                                                    
               found type `std::option::Option<hbbft::sync_key_gen::Ack>`                                                               
                                                                                                                                        
error[E0308]: mismatched types                                                                                                          
   --> src/hydrabadger/state.rs:191:21                                                                                                  
    |                                                                                                                                   
191 |                     Some(PartOutcome::Valid(ack)) => ack,                                                                         
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `std::result::Result`, found enum `std::option::Option`           
    |                                                                                                                                   
    = note: expected type `std::result::Result<hbbft::sync_key_gen::PartOutcome, hbbft::sync_key_gen::Error>`                           
               found type `std::option::Option<_>`                                                                                      
                                                                                                                                        
error[E0308]: mismatched types                                                                                                          
   --> src/hydrabadger/state.rs:192:21                                                                                                  
    |                                                                                                                                   
192 |                     Some(PartOutcome::Invalid(faults)) => panic!(                                                                 
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `std::result::Result`, found enum `std::option::Option`      
    |                                                                                                                                   
    = note: expected type `std::result::Result<hbbft::sync_key_gen::PartOutcome, hbbft::sync_key_gen::Error>`                           
               found type `std::option::Option<_>`                                                                                      
                                                                                                                                        
error[E0308]: mismatched types                                                                                                          
   --> src/hydrabadger/state.rs:197:21                                                                                                  
    |                                                                                                                                   
197 |                     None => unimplemented!(),                                                                                     
    |                     ^^^^ expected enum `std::result::Result`, found enum `std::option::Option`                                    
    |                                                                                                                                   
    = note: expected type `std::result::Result<hbbft::sync_key_gen::PartOutcome, hbbft::sync_key_gen::Error>`                           
               found type `std::option::Option<_>`                                                                                      
                                                                                                                                        
error[E0308]: mismatched types                                                                                                          
   --> src/hydrabadger/state.rs:201:63                                                                                                  
    |                                                                                                                                   
201 |                 ack_queue.as_ref().unwrap().push((*local_uid, ack.clone()));                                                      
    |                                                               ^^^^^^^^^^^ expected struct `hbbft::sync_key_gen::Ack`, found enum `std::option::Option`
    |                                                                                                                                   
    = note: expected type `hbbft::sync_key_gen::Ack`                                                                                    
               found type `std::option::Option<hbbft::sync_key_gen::Ack>`                                                               
                                                                                                                                        
error[E0308]: mismatched types                                                                                                          
   --> src/hydrabadger/state.rs:219:19                                                                                                  
    |                                                                                                                                   
219 |         Ok((part, ack))                                                                                                           
    |                   ^^^ expected struct `hbbft::sync_key_gen::Ack`, found enum `std::option::Option`                                
    |                                                                                                                                   
    = note: expected type `hbbft::sync_key_gen::Ack`                                                                                    
               found type `std::option::Option<hbbft::sync_key_gen::Ack>`                                                               
                                                                                                                                        
error: aborting due to 12 previous errors                                                                                               
                                                                                                                                        
Some errors occurred: E0277, E0308, E0599.                                                                                              
For more information about an error, try `rustc --explain E0277`.                                                                       
error: Could not compile `hydrabadger`.                                                                                                 

To learn more, run the command again with --verbose.

```

--- a/src/hydrabadger/handler.rs
+++ b/src/hydrabadger/handler.rs
@@ -14,7 +14,7 @@ use hbbft::{
     crypto::{PublicKey, PublicKeySet},
     dynamic_honey_badger::{ChangeState, JoinPlan, Message as DhbMessage, Change as DhbChange, Input as DhbInput},
     // queueing_honey_badger::{Change as QhbChange, Input as QhbInput},
-    sync_key_gen::{Ack, Part, PartOutcome, SyncKeyGen},
+    sync_key_gen::{Ack, AckOutcome, Part, PartOutcome, SyncKeyGen},
     DistAlgorithm, Target
 };
 use peer::Peers;
@@ -193,11 +193,11 @@ impl<T: Contribution> Handler<T> {
         ack_count: &mut usize,
     ) {
         info!("KEY GENERATION: Handling ack from '{}'...", uid);
-        let fault_log = sync_key_gen.handle_ack(uid, ack.clone());
-        if !fault_log.is_empty() {
-            error!("Errors handling ack: '{:?}':\n{:?}", ack, fault_log);
+        let ack_outcome = sync_key_gen.handle_ack(uid, ack.clone()).expect("Failed to handle Ack.");
+        match ack_outcome {
+            AckOutcome::Invalid(fault) => error!("Error handling ack: '{:?}':\n{:?}", ack, fault),
+            AckOutcome::Valid => *ack_count += 1,
         }
-        *ack_count += 1;
     }
 
     fn handle_queued_acks(
@@ -234,14 +234,18 @@ impl<T: Contribution> Handler<T> {
                 let mut rng = rand::OsRng::new().expect("Creating OS Rng has failed");
                 let mut skg = sync_key_gen.as_mut().unwrap();
                 let ack = match skg.handle_part(&mut rng, src_uid, part) {
-                    Some(PartOutcome::Valid(ack)) => ack,
-                    Some(PartOutcome::Invalid(faults)) => panic!(
+                    Ok(PartOutcome::Valid(Some(ack))) => ack,
+                    Ok(PartOutcome::Invalid(faults)) => panic!(
                         "Invalid part \
                          (FIXME: handle): {:?}",
                         faults
                     ),
-                    None => {
+                    Ok(PartOutcome::Valid(None)) => {
                         error!("`DynamicHoneyBadger::handle_part` returned `None`.");
+                        return;
+                    }
+                    Err(err) => {
+                        error!("Error handling Part: {:?}", err);
                         return;
                     }
                 };

--- a/src/hydrabadger/state.rs
+++ b/src/hydrabadger/state.rs
@@ -187,14 +187,15 @@ impl<T: Contribution> State<T> {
                 part = opt_part.expect("This node is not a validator (somehow)!");
 
                 info!("KEY GENERATION: Handling our own `Part`...");
-                ack = match sync_key_gen.handle_part(&mut rng, &local_uid, part.clone()) {
-                    Some(PartOutcome::Valid(ack)) => ack,
-                    Some(PartOutcome::Invalid(faults)) => panic!(
+                ack = match sync_key_gen.handle_part(&mut rng, &local_uid, part.clone())
+                                        .expect("Handling our own Part has failed") {
+                    PartOutcome::Valid(Some(ack)) => ack,
+                    PartOutcome::Invalid(faults) => panic!(
                         "Invalid part \
                          (FIXME: handle): {:?}",
                         faults
                     ),
-                    None => unimplemented!(),
+                    PartOutcome::Valid(None) => panic!("No Ack produced when handling Part."),
                 };
 
                 info!("KEY GENERATION: Queueing our own `Ack`...");

